src/components/Requirements/CompletedSubReqCourse.vue

```diff
@@ -87,7 +88,25 @@ export default Vue.extend({
       this.resetConfirmVisible = true;
     },
     onResetConfirmClosed(isReset: boolean): void {
-      if (isReset) deleteCourseFromSemesters(this.courseTaken.uniqueId, this.$gtag);
+      if (isReset) {
+        if (this.isTransferCredit) {
+          const type = this.courseTaken.code.substr(0, 2);
+          const name = this.courseTaken.code.substr(3);
+
+          const onBoardingData = store.state.onboardingData;
+
+          db.batch()
+            .set(onboardingDataCollection.doc(store.state.currentFirebaseUser.email), {
+              tookSwim: onBoardingData.tookSwim,
+              colleges: [{ acronym: onBoardingData.college }],
+              class: onBoardingData.transferCourse,
+              majors: onBoardingData.major.map(m => ({ acronym: m })),
+              minors: onBoardingData.minor.map(m => ({ acronym: m })),
+              exam: onBoardingData.exam.filter(e => !(e.type === type && e.subject === name)),
+            } as FirestoreOnboardingUserData)
+            .commit();
```

Comment:
You don't really need batch here, since you are only updating one document here. You can also use `.update` so that you don't have to copy over the unchanged fields like `majors, minors`